lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,459 @@
+                    err_string.format("intense precip", this_time))
+            # Increase prob(lightning) to Risk 2 (pl_dict[2]) when
+            #   prob(precip > 7mm/hr) > phighthresh
+            cube_slice.data = np.where(
+                (high_precip.data >= self.phighthresh) &
+                (cube_slice.data < self.pl_dict[2]),
+                self.pl_dict[2], cube_slice.data)
+            # Increase prob(lightning) to Risk 1 (pl_dict[1]) when
+            #   prob(precip > 35mm/hr) > ptorrthresh
+            cube_slice.data = np.where(
+                (torr_precip.data >= self.ptorrthresh) &
+                (cube_slice.data < self.pl_dict[1]),
+                self.pl_dict[1], cube_slice.data)
+
+            # Decrease prob(lightning) where prob(precip > 0) is low.
+            cube_slice.data = apply_double_scaling(
+                this_precip, cube_slice, self.precipthr, self.ltngthr)
+
+            new_cube_list.append(cube_slice)
+
+        new_cube = new_cube_list.merge_cube()
+        new_cube = check_cube_coordinates(
+            prob_lightning_cube, new_cube)
+        return new_cube
+
+    def apply_ice(self, prob_lightning_cube, ice_cube):
+        """
+        Modify Nowcast of lightning probability with ice data from radarnet
+        composite (VII; Vertically Integrated Ice)
+
+        Args:
+            prob_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability.
+                The forecast_period coord is modified in-place to "minutes".
+            ice_cube (iris.cube.Cube):
+                Analysis of vertically integrated ice (VII) from radar
+                thresholded at 0.5, 1.0, 2.0 kg m^-2
+                Units of threshold coord modified in-place to kg m^-2
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing updated nowcast lightning probability.
+                This cube will have the same dimensions and meta-data as
+                prob_lightning_cube.
+                The influence of the data in ice_cube reduces linearly to zero
+                as forecast_period increases to 2H30M.
+
+        Raises:
+            iris.exceptions.ConstraintMismatchError:
+                If ice_cube does not contain the expected thresholds.
+        """
+        prob_lightning_cube.coord('forecast_period').convert_units('minutes')
+        # check prob-ice threshold units are as expected
+        ice_cube.coord('threshold').convert_units('kg m^-2')
+        new_cube_list = iris.cube.CubeList([])
+        err_string = "No matching prob(Ice) cube for threshold {}"
+        for cube_slice in prob_lightning_cube.slices_over('time'):
+            fcmins = cube_slice.coord('forecast_period').points[0]
+            for threshold, prob_max in zip(self.ice_thresholds,
+                                           self.ice_scaling):
+                ice_slice = ice_cube.extract(
+                    iris.Constraint(threshold=threshold))
```

Comment:
Is there no limit to the lead times over which the VII observation is valid?

Reply:
Yes, but not here. See line that defines "ice_scaling" as a function of fcmins. As ice_cube is an analysis, time will be a scalar coordinate.